pkg/provider/provider.go

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package provider
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/crypto"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/trustbloc/edge-core/pkg/storage"
+)
+
+// Provider represents provider with functionality needed for keystore.
+type Provider interface {
```

Comment:
I wonder if there's another name we can come up with for this interface to make it more distinguishable from other "Provider" types

Reply:
addressed it in the comment above (will consider to come up with a better name in upcoming refactoring)

Reply:
As an example: in EDV, I have an "edvprovider" that extends a provider to provide additional functionality. Perhaps a similar naming scheme can work here